### ISSUE TITLE
Keyboard does not provide PageUp / PageDown

### ISSUE THREAD

**Issue by boroli** [maliit/keyboard#54]
The maliit keyboard in Plasma Mobile provides some control keys in the upper row. Is it possible to get the PgUp / PgDn keys to work? 

Or is there any one click workaround?

The Squeekyboard in Phosh has now a nice feature to access them (PgUp, PgDn, Home, End), just by clicking the Shift key. I found that solution very handy.

**Comment by tusooa** [comment]
Control keys in the upper row? You mean in the terminal?
That belongs to the terminal not maliit. If you slide down to dismiss the keyboard, the "control keys" will persist.
Also see https://invent.kde.org/plasma-mobile/qmlkonsole/-/blob/master/src/contents/ui/TerminalPage.qml .

**Comment by boroli** [comment]
Oh I see. Thanks for the link. That looks pretty hard coded.

Are there any plans to add control keys directly to maliit in the future?

**Comment by boroli** [comment]
I opened also a new issue upstream:

https://invent.kde.org/plasma-mobile/qmlkonsole/-/issues/7

**Comment by tusooa** [comment]
> Oh I see. Thanks for the link. That looks pretty hard coded.
> 
> Are there any plans to add control keys directly to maliit in the future?

I guess they are willing to accept mrs about this but I am not aware of any plans of them to do anything...

**Comment by kugiigi** [comment]
I believe Rodney already upstreamed most things from Ubuntu Touch's maliit fork and it includes buttons for home, end, etc. They are available by long pressing the space bar or swiping from the bottom in UT's fork.
Here's a sample how it looks on UT. Ignore the left and right column. Only the top row is actually implemented 😄
![screenshot20220330_010901242](https://user-images.githubusercontent.com/29456786/160667456-c0662c6e-0f2c-4bea-890d-7b53b0d78ca1.png)

**Comment by dobey** [comment]
> Here's a sample how it looks on UT

No. That's how it looks on your device where you've been messing around with things.

Here's how it looks on Plasma Mobile. And no, it doesn't have page up/dn keys.

![Screenshot from 2022-03-29 15-32-47](https://user-images.githubusercontent.com/6557070/160692803-0236aff8-3db9-4561-80c8-daa31e4b140f.png)

![Screenshot from 2022-03-29 15-29-54](https://user-images.githubusercontent.com/6557070/160692520-1163ca76-8389-449f-9083-ed187c4954f6.png)

**Comment by plata** [comment]
Can we use this issue as a more general "how to handle non-character keys"?

Use case: navigation in terminal emulator (tab, up/down, page up/down) + modifiers (ctrl, alt). Switching to the spacebar menu for this is too annoying. Yes, work-arounds can be implemented in the application (e.g. Konsole) but is this really a good solution? Those keys are also part of the physical keyboard so it's understandable that people expect it in Maliit. Moreover, it would unify the UX across different terminal emulators.
